utils/dynamicLink.js

```diff
@@ -13,9 +13,24 @@ const generateLongDynamicLink = async (redirectDesktopLink, redirectAndroidLink
     if (redirectAndroidLink === null) redirectAndroidLink = redirectDesktopLink
     if (redirectIosLink === null) redirectIosLink = redirectDesktopLink
 
-    const longDynamicLink = `${FIREBASE_DYNAMIC_LINK_URL}?&apn=${FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE}&afl=${redirectAndroidLink}&isi=${BETTER_APP_STORE_ID}&ibi=${FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE}&ifl=${redirectIosLink}&efr=1&ofl=${redirectDesktopLink}`
+    const longDynamicLink = `${FIREBASE_DYNAMIC_LINK_URL}?&apn=${FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE}&afl=${redirectAndroidLink}&isi=${BETTER_APP_STORE_ID}&ibi=${FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE}&ifl=${redirectIosLink}&efr=1&ofl=${redirectDesktopLink}&d=1`
+
     const { shortLink } = await firebaseDynamicLinks.createLink({
-        longDynamicLink
+        dynamicLinkInfo: {
+            domainUriPrefix: `${FIREBASE_DYNAMIC_LINK_URL}`,
+            link: longDynamicLink,
+            androidInfo: {
+                androidPackageName: FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE,
+                androidFallbackLink: redirectAndroidLink
+            },
+            iosInfo: {
+                iosBundleId: FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE,
+                iosFallbackLink: redirectIosLink
+            }
+        },
+        suffix: {
+            option: 'SHORT'
+        }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new implementation of the `createLink` method is more readable and maintainable than the previous one. However, there's a potential issue with the `link` property in the `dynamicLinkInfo` object. The `link` property should be the link that you want to open when the dynamic link is clicked, not the long dynamic link itself. 

```diff
-    link: longDynamicLink,
+    link: YOUR_DESIRED_LINK,
```

Please replace `YOUR_DESIRED_LINK` with the actual URL you want to open when the dynamic link is clicked.

Also, it seems like you're still generating the `longDynamicLink` but not using it anywhere. If it's not needed anymore, consider removing it to clean up the code.

```diff
-    const longDynamicLink = `${FIREBASE_DYNAMIC_LINK_URL}?&apn=${FIREBASE_DYNAMIC_LINK_ANDROID_APP_PACKAGE}&afl=${redirectAndroidLink}&isi=${BETTER_APP_STORE_ID}&ibi=${FIREBASE_DYNAMIC_LINK_IOS_APP_PACKAGE}&ifl=${redirectIosLink}&efr=1&ofl=${redirectDesktopLink}&d=1`
```

If you still need the `longDynamicLink`, please ignore the second suggestion.


<!-- This is an auto-generated comment by OSS CodeRabbit -->